src/schema/v2/infiniteDiscovery/discoverArtworks.ts

```diff
@@ -47,21 +49,76 @@ export const DiscoverArtworks: GraphQLFieldConfig<void, ResolverContext> = {
         },
       }),
     },
+    useRelatedArtworks: { type: GraphQLBoolean, defaultValue: false },
   }),
   resolve: async (
     _root,
     args,
-    { weaviateCreateObjectLoader, weaviateGraphqlLoader, artworksLoader }
+    {
+      weaviateCreateObjectLoader,
+      weaviateGraphqlLoader,
+      artworksLoader,
+      relatedArtworksLoader,
+      marketingCollectionLoader,
+      savedArtworksLoader,
+    }
   ) => {
     if (
+      !artworksLoader ||
       !weaviateGraphqlLoader ||
-      !weaviateCreateObjectLoader ||
-      !artworksLoader
+      !weaviateCreateObjectLoader
     ) {
       new Error("A loader is not available")
     }
 
-    const { userId, limit = 5, offset = 0, certainty = 0.5, sort } = args
+    const {
+      userId,
+      limit = 5,
+      offset = 0,
+      certainty = 0.5,
+      sort,
+      useRelatedArtworks,
+    } = args
+
+    if (useRelatedArtworks) {
+      if (!savedArtworksLoader) {
+        return new Error("Saved artworks loader is not available")
+      }
+
+      const { body: savedArtworks } = await savedArtworksLoader({
+        size: 28,
+        sort: "-position",
+        user_id: userId,
+        private: true,
+      })
+
+      // Extract saved artwork IDs
+      const savedArtworkIds = savedArtworks.map((artwork) => artwork.id)
+
+      // Load curated artworks
+      const curatedArtworks = await marketingCollectionLoader("curators-picks")
+      const curatedArtworkIds = curatedArtworks?.artwork_ids || []
+
+      // Select two random artwork IDs from curated artworks
+      const randomCuratedArtworkIds = sampleSize(curatedArtworkIds, 2)
+
+      // Combine IDs: use curated if no saved artworks, otherwise combine
+      const finalArtworkIds =
+        savedArtworkIds.length > 0
+          ? [...savedArtworkIds, ...randomCuratedArtworkIds]
```

Comment:
My understanding is that we don't want the two random artworks to be included list of IDs that are used to return the "similar artworks". Rather we want to inject them into the recommended artworks at the end, to provide some "challenge" works.

Reply:
Yeah! That's a good call out!